build-version.sh

```diff
@@ -30,6 +30,20 @@ if [ "$TARGET_ARCH" == "x86_64-unknown-linux-musl" ]; then
 
     CARGO_BIN_DIR="${TEMPDIR}/cargo/bin"
     CRATES2_JSON_PATH="${TEMPDIR}/cargo/.crates2.json"
+elif [ "$TARGET_ARCH" == "aarch64-unknown-linux-gnu" ]; then
+    wget $(curl https://ziglang.org/download/index.json | jq 'to_entries | map([.key, .value])[1][1]["x86_64-linux"] | .tarball' | sed -e 's/^"//' -e 's/"$//') -O zig
+    mkdir zigfolder
+    tar -xf ./zig -C zigfolder --strip-components 1
+    rm zig
+    sudo mv zigfolder/zig /usr/local/bin
+    rustup target add "$TARGET_ARCH"
+    echo "[target.aarch64-unknown-linux-gnu]" >> ~/.cargo/config
+    echo '["linker = "./zig-aarch64.sh"]' >> ~/.cargo.config
```

Comment:
I think we should instead maintain a new file `cargo-config` since using `echo` to create a file is basically unreadable.

The file should look like this according to [this SSO](https://stackoverflow.com/questions/62371337/rust-cargo-toml-specify-custom-path-for-c-linker-and-compiler):

```
[target.aarch64-unknown-linux-gnu]
linker = "./zig-aarch64.sh"
ar = "./zig-aarch64.shr"
```

Reply:
hmm, i think for now it is fine.

Reply:
i was going off of https://actually.fyi/posts/zig-makes-rust-cross-compilation-just-work/

Reply:
I think that's partially because:

```
echo '["linker = "./zig-aarch64.sh"]'
```

prints:

```
["linker = "./zig-aarch64.sh"]
```

which is not what cargo expected.

Reply:
I updated it in a recent commit but it's still broken